src/Typecheck.elm

```diff
@@ -151,6 +152,12 @@ typecheck env t =
         typecheckLam env t
 
       App x y -> typecheckApp env t
+
+      Tuple a b ->
+        case (typecheck env a, typecheck env b) of
+          (Checks x, Checks y) -> Checks (TTuple x y)
+          _ -> Invalid
```

Comment:
Even if typecheck env a or typecheck env b return `Partial _` or `Fails _ _ _ _` we will return Invalid. Is this the intended behavior? Wouldn't we want to cascade a single (or double) Partial or Fails in a tuple to the result of typechecking that tuple?

Reply:
You are right, it should cascade Partial or Fails. I'll work on this

Reply:
I went ahead and changed the return types according to the typecheck of each member of the tuple. There are 12 different cases, and I was a little unsure of how to deal with them. At the end I implemented it according to the following table, which we can discuss and easily change if need to be:

Followed this convention (we can discuss this)
(x,y)
x\y         | Checks  | Fails    | Partial   | Invalid
-----------------------------------------------------
Checks | Checks | Fails     | Partial   | Invalid
Fails      | Fails      | Fails     | Fails      | Invalid
Partial   | Partial   | Fails     | Partial   | Invalid
Invalid   | Invalid   | Invalid  | Invalid   | Invalid

Reply:
I will wait to merge this to master until we have discussed if the vtypes are appropriate